### DISCUSSION TITLE
What is the owner parameter for?

### DISCUSSION THREAD

**Discussion by SayakMukhopadhyay** [community/community#24509] [Apps, API and Webhooks]
<p>Hi</p>
<p>I can see that the <code>owner</code> field is used in multiple places as a parameter or in the returned data. Also some endpoints like the <code>teams</code> endpoint require the <code>org</code> field. What is the significance of the <code>owner</code> field. If I had to guess I would say that the owner refers to the account (organisation or user) that that particular resource belongs to in the url. Am I correct?</p>
<p>Also, how do I get the owners from an <code>installation</code> resource. Right now, I can see that the <code>installation</code> resource returns an <code>account</code> object which has a <code>login</code> field which looks very much like the username or organisation name used in the urls. Is it fine to use that value as the <code>owner</code> value?</p>

**Comment by francisfuzz** [comment]
<aside class="quote no-group" data-username="sayakmukhopadhyay" data-post="1" data-topic="120098">
<div class="title">
<div class="quote-controls"></div>
 sayakmukhopadhyay:</div>
<blockquote>
<p>I can see that the <code>owner</code> field is used in multiple places as a parameter or in the returned data. Also some endpoints like the <code>teams</code> endpoint require the <code>org</code> field. What is the significance of the <code>owner</code> field. If I had to guess I would say that the owner refers to the account (organisation or user) that that particular resource belongs to in the url. Am I correct?</p>
</blockquote>
</aside>
<p><a class="mention" href="/sayakmukhopadhyay">@sayakmukhopadhyay</a> - That’s correct! :+1:</p>
<aside class="quote no-group" data-username="sayakmukhopadhyay" data-post="1" data-topic="120098">
<div class="title">
<div class="quote-controls"></div>
 sayakmukhopadhyay:</div>
<blockquote>
<p>Also, how do I get the owners from an <code>installation</code> resource. Right now, I can see that the <code>installation</code> resource returns an <code>account</code> object which has a <code>login</code> field which looks very much like the username or organisation name used in the urls. Is it fine to use that value as the <code>owner</code> value?</p>
</blockquote>
</aside>
<p>This is a great question. Before sharing any specific advice, I’m wondering which endpoint are you asking about, and which specific resource are you interested in that would require the use of the name of the account as the <code>owner</code> value?</p>
<p>cc: <a href="https://docs.github.com/en/rest/reference/apps">https://docs.github.com/en/rest/reference/apps</a></p>

**Comment by SayakMukhopadhyay** [comment]
<aside class="quote group-github-employees" data-username="francisfuzz" data-post="2" data-topic="120098">
<div class="title">
<div class="quote-controls"></div>
 francisfuzz:</div>
<blockquote>
<p>I’m wondering which endpoint are you asking about, and which specific resource are you interested in that would require the use of the name of the account as the <code>owner</code> value?</p>
</blockquote>
</aside>
<p>I using the <code>installation.created</code> webhook <a href="https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#installation" rel="nofollow noopener">https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#installation</a>. Essentially, I would like to get the reference to the organisation/user account where the installation of my bot is happening so that I can call other queries which needs the organisation name like to get the content of a repository <a href="https://docs.github.com/en/rest/reference/repos#get-repository-content" rel="nofollow noopener">https://docs.github.com/en/rest/reference/repos#get-repository-content</a>. I was wondering if I can use the <code>account.login</code> field from the <code>installation.created</code> webhook in the <code>owner</code> field of the <code>get-repository-content</code> API call.</p>

**Comment by francisfuzz** [comment, marked as answer]
<aside class="quote no-group" data-username="sayakmukhopadhyay" data-post="3" data-topic="120098">
<div class="title">
<div class="quote-controls"></div>
 sayakmukhopadhyay:</div>
<blockquote>
<p>I was wondering if I can use the <code>account.login</code> field from the <code>installation.created</code> webhook in the <code>owner</code> field of the <code>get-repository-content</code> API call.</p>
</blockquote>
</aside>
<p><a class="mention" href="/sayakmukhopadhyay">@sayakmukhopadhyay</a> - Thanks for sharing that additional context! The value of <code>account.login</code> in that payload tells you who the owner of the repositories the application was just installed to.</p>
<p>However, knowing solely the owner is not enough to <a href="https://docs.github.com/en/rest/reference/repos#get-repository-content">get repository content</a>, since that endpoint also requires the name of the repository (<code>{repo}</code>) and the path (<code>{path}</code>).</p>
<p>It may be helpful to note here we have an <a href="https://docs.github.com/en/rest/reference/apps#installations">Installations API</a> that enables you to get information about the installations of your GitHub App and perform actions within those installations. In that API, there’s an endpoint for <a href="https://docs.github.com/en/rest/reference/apps#list-repositories-accessible-to-the-app-installation">listing repositories accessible to the app installation</a>. That can be helpful to determine which repositories your installation has access to at any given time before making any specific calls to any one repository. I hope this helps!</p>

**Comment by SayakMukhopadhyay** [comment]
<p>Hi <a class="mention" href="/francisfuzz">@francisfuzz</a><br>
Thanks for the clarification. I am already using the Installations API to get the list of repositories the installation has access to. My concern was reliably getting the owner info which as you clarified can be got from the <code>account.login</code> field. Thanks a lot for taking the time to help. Appreciate it!<br>
Regards</p>
